code/modules/urist/modules/wfc/modules/deepmaint/teleporters.dm

```diff
@@ -0,0 +1,151 @@
+
+/proc/deepmaint_yeet_trg(var/atom/movable/trg, var/xoffset = 1, var/yoffset = 1)
+	if(!ismovable(trg))
+		// This would be silly
+		return FALSE
+
+	if(isnull(trg.loc))
+		// No un-qdel-ing by accident!
+		return
+
+	if(ismob(trg) && !istype(trg, DEEPMAINT_TELEPORTABLE_MOBTYPE))
+		// Don't teleport AI eyes, ghosts, irate admins...
+		return
+
+	var/raw_pos_x = DEEPMAINT_RELATIVE_POS(trg.x, DEEPMAINT_VARIANT_SIZE_X, DEEPMAINT_VARIANT_SPACING_X) + DEEPMAINT_YEET_POS(0, xoffset, DEEPMAINT_VARIANT_SIZE_X, DEEPMAINT_VARIANT_SPACING_X)
+	var/raw_pos_y = DEEPMAINT_RELATIVE_POS(trg.y, DEEPMAINT_VARIANT_SIZE_Y, DEEPMAINT_VARIANT_SPACING_Y) + DEEPMAINT_YEET_POS(0, yoffset, DEEPMAINT_VARIANT_SIZE_Y, DEEPMAINT_VARIANT_SPACING_Y)
+
+	var/pos_x = raw_pos_x % world.maxx
+	var/pos_y = raw_pos_y % world.maxy
+	var/pos_z = trg.z
+
+	var/turf/outturf = locate(pos_x, pos_y, pos_z)
+
+	if(isnull(outturf))
+		return FALSE
+
+	trg.loc = outturf
+	return TRUE
+
+
+/proc/deepmaint_send_trg(var/atom/movable/trg)
+	if(!ismovable(trg))
+		// This would be silly
+		return FALSE
+
+	if(isnull(trg.loc))
+		// No un-qdel-ing by accident!
+		return
+
+	if(ismob(trg) && !istype(trg, DEEPMAINT_TELEPORTABLE_MOBTYPE))
+		// Don't teleport AI eyes, ghosts, irate admins...
+		return
+
+	var/tries = 1
+	var/success = FALSE
+
+	var/list/predicates = list(/proc/not_turf_contains_dense_objects)
+
+	while(tries --> 0)
+		var/turf/T = pick_area_turf(/area/map_template/deepmaint_wfc/alpha, predicates)
+
+		if(isnull(T))
+			continue
+
+		var/mob/M = trg
+		if(istype(M))
+			M.Stun(1)
+			M.flash_eyes(intensity = FLASH_PROTECTION_MAJOR * 10, override_blindness_check = TRUE, affect_silicon = TRUE, visual = FALSE, type = /obj/screen/fullscreen/blackout)
```

Comment:
Nothing major, just a little peeve. As these parameters are quite long, could we newline this for better readability?
```suggestion
			M.flash_eyes(
				intensity = FLASH_PROTECTION_MAJOR * 10, 
				override_blindness_check = TRUE,
				affect_silicon = TRUE,
				visual = FALSE,
				type = /obj/screen/fullscreen/blackout
			)
```

Reply:
Sure, more readability = more good